OpenEdXMobile/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -13,42 +13,112 @@
 import org.edx.mobile.model.course.CourseComponent;
 
 /**
- *  A central place for course data model transformation
+ * This class manages the caching mechanism of courses data.
  */
-
 @Singleton
 public class CourseManager {
     protected final Logger logger = new Logger(getClass().getName());
 
-    private final LruCache<String, CourseComponent> cachedComponent;
+    /**
+     * This count represents the maximum no of courses that will be cached via {@link LruCache}.
+     */
+    private final int NO_OF_COURSES_TO_CACHE = 15;
 
+    /**
+     * A cache to keep courses data in memory till ending of app session.
+     */
+    private final LruCache<String, CourseComponent> cachedComponent;
 
     @Inject
     CourseAPI courseApi;
 
     public CourseManager() {
-        cachedComponent = new LruCache<>(1);
+        cachedComponent = new LruCache<>(NO_OF_COURSES_TO_CACHE);
     }
 
+    public void doLruCacheOfCourse(@NonNull String courseId,
```

Comment:
This is an implementation detail that the CourseManager is using LRU cache at the backend to cache a course.

This function's name should be generic, like `putInCache` or `cacheCourse`.